src/main/scala/shine/OpenCL/Module.scala

```diff
@@ -0,0 +1,38 @@
+package shine.OpenCL
+
+case class Module(host: shine.C.Module, kernels: Seq[KernelModule]) {
+  def compose(other: Module): Module =
+    Module(
+      host compose other.host,
+      kernels ++ other.kernels)
+}
+
+object Module {
+  def compose(ms: Seq[Module]): Module = ms.reduce(_ compose _)
+
+  def translateToString(m: Module): String =
+    s"""
+       |${m.kernels.map { km =>
+      // assumes a single kernel per module
```

Comment:
I am not sure if we actually want this long-term? it might actually be beneficial to bundle all the kernels into a single module so that common definitions can be shared instead of duplicated? This would require changing the runtime and codegen slightly, so this was the easiest short-term solution.

Reply:
I guess my point was, that we will have to think about this and touch this piece of code then anyway.
Currently the types suggest that we could support multiple kernels, but we clearly don't (like here).

I am fine keeping it as a comment here.